src/_premake_init.lua

```diff
@@ -1113,6 +1111,40 @@
 		linktimeoptimization("Default")
 	end)
 
+	api.deprecateValue("flags", "FatalWarnings", "Use `fatalwarnings` instead.",
+	function(value)
+		fatalwarnings({ "Compile", "Link" })
+	end,
+	function(value)
+		removefatalwarnings({ "Compile", "Link" })
+	end)
+
+	api.deprecateValue("flags", "FatalCompileWarnings", "Use `fatalwarnings` instead.",
+	function(value)
+		fatalwarnings({ "Compile" })
+	end,
+	function(value)
+		removefatalwarnings({ "Compile" })
+	end)
+
+	api.deprecateValue("flags", "FatalLinkWarnings", "Use `fatalwarnings` instead.",
+	function(value)
+		fatalwarnings({ "Link" })
+	end,
+	function(value)
+		removefatalwarnings({ "Link" })
+	end)
+
+	premake.filterFatalWarnings = function(tbl)
+		if type(tbl) == "table" then
+			return table.filter(tbl, function(warning)
+				return not (warning == "Compile" or warning == "Link")
+			end)
+		else
+			return tbl
+		end
+	end
```

Comment:
Is this the best location for this?

Similarly, I had a thought that we could also do something during the bake stage to remove and convert compile and link to something like `cfg._fatalcompilewarnings = true` and `cfg._fatallinkwarnings = true`. It would remove any mistakes around using `fatalwarnings` directly (especially in external modules), but it is probably a bit more complicated to make it work like this.

Reply:
Let's circle back on figuring out a robust way of doing something like this in the bake process. I like the idea, though.